op-challenger/game/fault/trace/vm/op_program_server_executor.go

```diff
@@ -33,11 +33,11 @@ func (s *OpProgramServerExecutor) OracleCommand(cfg Config, dataDir string, inpu
 	if inputs.L2Head != (common.Hash{}) {
 		args = append(args, "--l2.head", inputs.L2Head.Hex())
 	}
-	if inputs.L2OutputRoot != (common.Hash{}) {
-		args = append(args, "--l2.outputroot", inputs.L2OutputRoot.Hex())
-	}
+	// --l2.agreed-prestate and --l2.outputroot are mutually exclusive flags
```

Comment:
I'd suggest we stop setting L2OutputRoot in https://github.com/ethereum-optimism/optimism/blob/83d862591eaca4b41dc782944d9e1deb3c9339d8/op-challenger/game/fault/trace/super/super_cannon.go#L46

The original idea was to keep this option translation as simple as possible so that it's predictable. I'm not against this change as well, but I think fixing the upstream is important so we don't have a value being passed around that is actually intended to be ignored.

Reply:
Acknowledged. I'll remove it.

Reply:
ehhh, it feels odd to me that the provider is concerned keeping both fields mutually exclusive.
But it's a weakly held opinion.

Reply:
Changed to the old behavior, with properly setup GameInputs

Reply:
Yeah I'm fairly weakly opinionated about it as well. Bike-shedding: we maybe should have `NewInteropGameInputs` and `NewPreInteropGameInputs` functions to force things to be set correctly but I think it's just shifting the question of which code knows the rules to a third place...

Frankly I'm looking forward to just removing the pre-interop support from challenger so these things aren't confusing.

Reply:
how about introducing a "super cannon executor" that knows not to supply `--l2.outputroot`? Similar to kon's https://github.com/ethereum-optimism/optimism/blob/develop/op-challenger/game/fault/trace/vm/kona_super_server_executor.go

Reply:
Yeah that's another option too. Would work well if it could use a separate `LocalGameInputs` that just doesn't have that field.  If `cannon.NewTraceProvider` just accepted the `executor` instead of building it it would work, since the local inputs are only passed through to the executor.